File: clean_beauty.py
import re

with open('src/components/LiveStudioModal.jsx', 'r') as f:
    content = f.read()

# 1. Remove import
content = re.sub(r"import AiFaceEffectOverlay from '\./Overlays/AiFaceEffectOverlay';\n?", '', content)

# 2. Remove states 
content = re.sub(r"  const \[beautyFilter[\s\S]*?const \[beautySubTab[\s\S]*?\];\n?", '', content)

# 3. Remove AiFaceEffectOverlay component usage
content = re.sub(r"            \{\/\* Real-time AI Face & AR Overlay \*\/\}[\s\S]*?\/>\n", '', content)

# 4. Remove 'beauty' drawer tab logic in Top Bar menu
content = re.sub(r"                    \{\/\* Effects & Filters \*\/\}[\s\S]*?\{\/\* Tools & Moderation \*\/}", '{/* Tools & Moderation */}', content)

# 5. Remove Beauty & AR Filters button on main UI
content = re.sub(r"            \{\/\* Beauty & AR Filters \*\/\}[\s\S]*?<\/button>\n", '', content)

# 6. Remove the drawer header item for beauty
content = re.sub(r"                  \{activeTabDrawer === 'beauty' && window\.loc\('✨ زیبایی', '✨ Beauty'\)\}\n?", '', content)

# 7. Remove the whole BEAUTY & AR EFFECTS STUDIO DRAWER
start_marker = "              {/* BEAUTY & AR EFFECTS STUDIO DRAWER */}"
end_marker = "                </div>\n              )}\n"
# Find start index
start_idx = content.find(start_marker)
if start_idx != -1:
    # Find the next matching end index after start_idx
    # We can use a heuristic. Look for the end of the drawer block.
    # We know the block ends right before the closing of `activeTabDrawer && (`
    # Actually, it's easier to find the exact end block. Let's find the position of the next END marker.
    # The BEAUTY drawer contains "END LIVE CONFIRMATION MODAL" after it.
    end_live_confirm = content.find("{/* END LIVE CONFIRMATION MODAL */}")
    
    # Let's find the closing of the activeTabDrawer block
    # It looks like:
    #                 </div>
    #               )}
    #             </div>
    #           )}
    #       </div>
    #   )}
    # {/* ========================================================================= */}
    # {/* END LIVE CONFIRMATION MODAL */}
    
    search_str = "              )}\n            </div>\n          )}\n        </div>\n      )}\n\n      {/* ========================================================================= */}\n      {/* END LIVE CONFIRMATION MODAL */}"
    end_idx = content.find(search_str)
    
    if end_idx != -1:
        content = content[:start_idx] + content[end_idx:]

with open('src/components/LiveStudioModal.jsx', 'w') as f:
    f.write(content)

File: clean_beauty.js
const fs = require('fs');
let content = fs.readFileSync('src/components/LiveStudioModal.jsx', 'utf8');

// 1. Remove import
content = content.replace(/import AiFaceEffectOverlay from '\.\/Overlays\/AiFaceEffectOverlay';\n?/, '');

// 2. Remove states 112 to 129
// Find "const [beautyFilter, setBeautyFilter]" to "const [beautySubTab, setBeautySubTab]"
const stateRegex = /  const \[beautyFilter[\s\S]*?const \[beautySubTab[\s\S]*?\];\n?/m;
content = content.replace(stateRegex, '');

// 3. Remove AiFaceEffectOverlay component usage
const overlayRegex = /            \{\/\* Real-time AI Face & AR Overlay \*\/\}[\s\S]*?\/>\n/m;
content = content.replace(overlayRegex, '');

// 4. Remove 'beauty' drawer tab logic in Top Bar menu
const effectsMenuRegex = /                    \{\/\* Effects & Filters \*\/\}[\s\S]*?\{\/\* Tools & Moderation \*\/}/m;
content = content.replace(effectsMenuRegex, '{/* Tools & Moderation */}');

// 5. Remove Beauty & AR Filters button on main UI
const beautyBtnRegex = /            \{\/\* Beauty & AR Filters \*\/\}[\s\S]*?<\/button>\n/m;
content = content.replace(beautyBtnRegex, '');

// 6. Remove the drawer header item for beauty
content = content.replace(/                  \{activeTabDrawer === 'beauty' && window.loc\('✨ زیبایی', '✨ Beauty'\)\}\n?/m, '');

// 7. Remove the whole BEAUTY & AR EFFECTS STUDIO DRAWER
// This one is large. Starts with {/* BEAUTY & AR EFFECTS STUDIO DRAWER */}
const beautyDrawerRegex = /              \{\/\* BEAUTY & AR EFFECTS STUDIO DRAWER \*\/\}[\s\S]*?                  \)\} \/\* End Lighting \*\/\}[\s\S]*?                <\/div>\n              \)\}\n/m;

// wait, the end of the beauty drawer is:
//                 </div>
//               )}
// 
// Let's use a simpler regex for the beauty drawer by finding the start and an exact end string.

fs.writeFileSync('src/components/LiveStudioModal.jsx', content);
